exec/java-exec/src/test/java/org/apache/drill/exec/store/log/TestLogReader.java

```diff
@@ -389,4 +416,56 @@ public void testRawUMNoSchema() throws RpcException {
 
     RowSetUtilities.verify(expected, results);
   }
+
+  @Test
+  public void testProvidedSchema() throws Exception {
```

Comment:
Could you please add unit tests to check how this format plugin works with schema parameter in table function?
Example: `org.apache.drill.TestSchemaWithTableFunction`
We might need to check two cases:
`select * from table(t(schema=>'inline=(col1 varchar)'))`
`select * from table(t(type=>'logRegex', schema=>'inline=(col1 varchar)'))`

Reply:
Short answer: it does not seem to work. I tried this in the past and found that table functions take only simple values (numbers, strings), not lists. Since this plugin uses a list, I never could figure out how to use it with table functions. In particular, how would the table function know how to create the instance of `LogFormatField` within the list? Am I missing something?

This plugin, in particular, would very much benefit from the use of a table function so that the user does not have to define a new plugin config for each new file type.

If there is a way to make this work, we can add the test and describe the answer in the README file.

Reply:
`select * from table(t(schema=>'inline=(col1 varchar)'))` should work disregarding format properties. But log format has schema property so I am wondering if there will be a clash or schema parameter will be correctly resolved, since log format has it as list and schema provisioning in string.
Since now log format supports schema provisioning, the above query should apply schema for log files, could you please check this query?